projects/ngx-device-detector/src/lib/device-detector.service.ts

```diff
@@ -121,14 +132,21 @@ export class DeviceDetectorService {
    * if the current device is a mobile and also check current device is tablet so it will return false.
    * @returns whether the current device is a mobile
    */
-  public isMobile(userAgent = this.userAgent): boolean {
-    if (this.isTablet(userAgent)) {
-      return false;
+  public isMobile(userAgent = this.userAgent, checkOtherDevice = true): boolean {
+    if (this.storeMobile !== null && userAgent === this.userAgent) {
```

Comment:
Since we already have that calculated in the constructor, you can just do:
```typescript
return this.deviceType === DeviceType.Mobile
```
Notice that DeviceType is the enum I suggested above.

Reply:
we can not add below code because user can call that function with new userAgent
`return this.deviceType === DeviceType.Mobile`